api/src/middleware/authenticate.ts

```diff
@@ -39,7 +43,12 @@ const authenticate: RequestHandler = asyncHandler(async (req, res, next) => {
 		}
 
 		const user = await database
-			.select('role', 'directus_roles.admin_access', 'directus_roles.app_access')
+			.select(
+				'role',
+				'directus_roles.admin_access',
+				'directus_roles.app_access',
+				...userDynamicVars.map((v) => 'directus_users.' + v)
+			)
 			.from('directus_users')
```

Comment:
We might have to start relying on a UserService instance here so we can also reliably fetch nested relational fields coming from the user for the purposes of applying permissions 🤔

Reply:
Fetching nested relational fields sounds awesome, it opens up more flexibility for permissions. 👍🏼